Apps/W1/EDocumentConnectors/SignUp/app/src/Processing.Codeunit.al

```diff
@@ -0,0 +1,573 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.SignUp;
+
+using Microsoft.EServices.EDocument;
+using System.Telemetry;
+using System.Text;
+using System.Utilities;
+
+codeunit 6388 Processing
+{
+    Access = Internal;
+    Permissions = tabledata "E-Document" = m,
+                  tabledata "E-Document Service Status" = m;
+
+    procedure SendEDocument(var EDocument: Record "E-Document"; var TempBlob: Codeunit "Temp Blob"; var IsAsync: Boolean; var HttpRequest: HttpRequestMessage; var HttpResponse: HttpResponseMessage)
+    var
+        EDocumentServiceStatus: Record "E-Document Service Status";
+        EdocumentService: Record "E-Document Service";
+        FeatureTelemetry: Codeunit "Feature Telemetry";
+    begin
+        IsAsync := true;
+
+        EDocumentHelper.GetEdocumentService(EDocument, EdocumentService);
+        EDocumentServiceStatus.Get(EDocument."Entry No", EdocumentService.Code);
+
+        case EDocumentServiceStatus.Status of
+            EDocumentServiceStatus.Status::Exported:
+                SendEDocument(EDocument, TempBlob, HttpRequest, HttpResponse);
+            EDocumentServiceStatus.Status::"Sending Error":
+                if EDocument."Document Id" = '' then
+                    SendEDocument(EDocument, TempBlob, HttpRequest, HttpResponse);
+        end;
+
+        FeatureTelemetry.LogUptake('', ExternalServiceTok, Enum::"Feature Uptake Status"::Used);
```

Comment:
Nice!

Reply:
Ain't it :) However, I think a change is in place as 'ExternalServiceConnector' is a tad to generic. Should maybe be something like '<Vendor>ServiceConnector' --> 'SignUpServiceConnector'. We are not alone... What do you think? @Groenbech96

Reply:
You can look at the Avalara application for inspiration in most things. 
For this TOK use 'E-Document - SignUp'